provider/src/main/kotlin/org/gradle/kotlin/dsl/ActionExtensions.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl
+
+import org.gradle.api.Action
+
+
+/**
+ * Enables function invocation syntax on [Action] references.
```

Comment:
The function's existence enables this. But calling the function does not enable the ability to use the invocation syntax on action.
Consider changing the documentation to describe what the method is doing, not why this function exists.

Reply:
Thanks for the suggestion, @JLLeitschuh. I like how it is. Feel free to open a PR if you feel strongly about it.